### PR TITLE
OTAProxy refinement and test files implementation

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction
This PR introduces some refinement over the ota_proxy and test files for the ota_proxy.

## Major Changes
1. simple ORM for sqlite is introduced, making ota cache table definition more programmatic and easy to maintain,
2. refinement and optimization in ota_cache.py and db.py,
3. type hinting fix, improve type hints coverage,
4. introduce test files for ota_proxy(coverage: ~77%),
5. bump ota_cache db table definition to v3,
6. fix memory leak by tunning sqlite3 PRAGMA,

## NOTE
1. special file name scheme is covered at `test_ota_proxy_server.py`,
2. also in `test_ota_proxy_server.py`, a simple e2e tests is introduced, which 6 clients download ota_image via ota_proxy at the same time.

## Ticket
[T4PB-17047](https://tier4.atlassian.net/browse/T4PB-17047)
[T4PB-19287](https://tier4.atlassian.net/browse/T4PB-19287)

### CODE REVIEW

**Review by obi-t4 — COMMENTED** [review]
I left some minor comments.

File: tests/test_ota_proxy/test_ota_cache.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    def prepare_entries(self):
+        entries: Dict[str, CacheMeta] = {}
+        for target_size, rotate_num in cfg.BUCKET_FILE_SIZE_DICT.items():
+            for _i in range(rotate_num):
+                url = f"{target_size}#{_i}"
+                entries[url] = CacheMeta(
+                    url=url,
+                    bucket_idx=target_size,
+                    size=target_size,
+                    sha256hash=str(target_size),
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply tests/test_ota_proxy/test_ota_cache.py]: Updated in [5d520f8](https://github.com/tier4/ota-client/pull/171/commits/5d520f83db872b4fe326af9b858d2e2524714a43).

File: otaclient/app/common.py [hunk 2]
```diff
@@ -56,17 +56,14 @@ class OTAFileCacheControl(enum.Enum):
...
-        return set(input.split(","))
-
     @classmethod
     def parse_to_enum_set(cls, input: str) -> Set["OTAFileCacheControl"]:
-        _policies_set = cls.parse_to_value_set(input)
         res = set()
-        for p in _policies_set:
-            res.add(OTAFileCacheControl[p])
-
+        for policy in input.split(","):
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply otaclient/app/common.py]: Sure!

Updated in [5d520f8](https://github.com/tier4/ota-client/pull/171/commits/5d520f83db872b4fe326af9b858d2e2524714a43).

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**